dataset_conversion/convert_bids_to_nnUNetv2_region-based.py

```diff
@@ -0,0 +1,432 @@
+    niftis_dict = {
+        f"train": sorted(train_niftis),
+        f"test": sorted(test_nifitis)
+    }
+
+    # write the train and test niftis to a yaml file
+    with open(os.path.join(path_out, f"train_test_split_seed{args.seed}.yaml"), "w") as outfile:
+        yaml.dump(niftis_dict, outfile, default_flow_style=False)
+
+    # c.f. dataset json generation
+    # In nnUNet V2, dataset.json file has become much shorter. The description of the fields and changes
+    # can be found here: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md#datasetjson
+    # this file can be automatically generated using the following code here:
+    # https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/dataset_conversion/generate_dataset_json.py
+    # example: https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/dataset_conversion/Task055_SegTHOR.py
+
+    json_dict = OrderedDict()
+    json_dict['name'] = args.dataset_name
+    json_dict['description'] = args.dataset_name
+    json_dict['reference'] = "TBD"
+    json_dict['licence'] = "TBD"
+    json_dict['release'] = "0.0"
+    json_dict['numTraining'] = train_ctr
+    json_dict['numTest'] = test_ctr
+    json_dict['seed_used'] = args.seed
+    json_dict['dataset_versions'] = dataset_commits
+    json_dict['image_orientation'] = "RPI"
+
+    # The following keys are the most important ones.
+    """
+    channel_names:
+        Channel names must map the index to the name of the channel. For BIDS, this refers to the contrast suffix.
+        {
+            0: 'T1',
+            1: 'CT'
+        }
+    Note that the channel names may influence the normalization scheme!! Learn more in the documentation.
+
+    labels:
+        This will tell nnU-Net what labels to expect. Important: This will also determine whether you use region-based training or not.
+        Example regular labels:
+        {
+            'background': 0,
+            'left atrium': 1,
+            'some other label': 2
+        }
+        Example region-based training: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/region_based_training.md
+        {
+            'background': 0,
+            'whole tumor': (1, 2, 3),
+            'tumor core': (2, 3),
+            'enhancing tumor': 3
+        }
+        Remember that nnU-Net expects consecutive values for labels! nnU-Net also expects 0 to be background!
+    """
+
+    json_dict['channel_names'] = {
+        0: "acq-ax_T2w",
+    }
+
+    if not args.region_based:
+        json_dict['labels'] = {
+            "background": 0,
+            "lesion": 1,
+        }
+    else:
+        json_dict['labels'] = {
+            "background": 0,
+            "sc": [1, 2],
+            # "sc": 1,
+            "lesion": 2,
+        }
+        json_dict['regions_class_order'] = [1, 2]
+
+    # Needed for finding the files correctly. IMPORTANT! File endings must match between images and segmentations!
+    json_dict['file_ending'] = ".nii.gz"
+
```

Comment:
Just to be consistent with latest findings about axes swaps -- maybe we should specify the IO reader ? 

```
json_dict['overwrite_image_reader_writer'] = "NibabelIO"
```